tests/models/grounding_dino/test_modeling_grounding_dino.py

```diff
@@ -54,6 +56,39 @@
     from transformers import AutoProcessor
 
 
+def generate_fake_bounding_boxes(n_boxes):
+    """Generate bounding boxes in the format (cx, cy, w, h)"""
+    # Validate the input
+    if not isinstance(n_boxes, int):
+        raise ValueError("n_boxes must be an integer")
+    if n_boxes <= 0:
+        raise ValueError("n_boxes must be a positive integer")
+
+    # Generate random bounding boxes in the format (cx, cy, w, h)
+    bounding_boxes = torch.rand((n_boxes, 4))
+
+    # Extract the components
+    cx = bounding_boxes[:, 0]
+    cy = bounding_boxes[:, 1]
+    w = bounding_boxes[:, 2]
+    h = bounding_boxes[:, 3]
+
+    # Ensure width and height do not exceed bounds
+    w = torch.min(w, torch.tensor(1.0))
+    h = torch.min(h, torch.tensor(1.0))
+
+    # Ensure the bounding box stays within the normalized space
+    cx = torch.where(cx - w / 2 < 0, w / 2, cx)
+    cx = torch.where(cx + w / 2 > 1, 1 - w / 2, cx)
+    cy = torch.where(cy - h / 2 < 0, h / 2, cy)
+    cy = torch.where(cy + h / 2 > 1, 1 - h / 2, cy)
+
+    # Combine back into bounding boxes
+    bounding_boxes = torch.stack([cx, cy, w, h], dim=1)
```

Comment:
nit, even if it is a tests let's avoid ane letter variable names, please use `center_x`, `center_y`, `height`, and `width`

Reply:
done